File: nutrinova-api/NutrinovaData/RequestModels/EditRecipeRequestModel.cs
public class EditRecipeRequestModel
{
  public Guid? Id { get; set; }

  public string? Description { get; set; }

  public string? Notes { get; set; }

  public List<string>? Tags { get; set; }

  public decimal Amount { get; set; }

  public int UnitId { get; set; }

  public UnitRequestModel ServingsUnit { get; set; } = null!;

  public List<EditFoodRequestModel> RecipeFoods { get; set; } = new List<EditFoodRequestModel>();
}

File: nutrinova-ui/src/app/(authorized)/recipes/edit/_models/EditRecipeRequestModel.ts
import { UnitOption } from "@/app/(authorized)/food/_models/unitOption";

export interface EditRecipeRequestModel {
  id: string;
  description?: string;
  tags?: string[];
  notes?: string;
  recipeFoods: EditRecipeFoodRequestModel[];
  amount?: number;
  note?: string;
  servingsUnit?: UnitOption | null;
  unitId?: number;
  categoryId: number;
}

export interface EditRecipeFoodRequestModel {
  id: string;
  name: string;
  servingSize: number;
  unitId: number;
  unitName: string;
  unit?: UnitOption;
  servingSizeUnitName?: string;
}